Library/Homebrew/utils/analytics.rb

```diff
@@ -69,12 +70,53 @@ def report(type, metadata = {})
         nil
       end
 
+      def report_influx(category, metadata = {})
+        curl = Utils::Curl.curl_executable
+        args = []
+        tags = []
+
+        metadata[:version] = HOMEBREW_VERSION
+        metadata[:product] = HOMEBREW_PRODUCT
+        metadata.each do |key, value|
+          next unless key
+          next unless value
+
+          tags << "#{key}=#{value.sub(" ", "\\ ")}"
+        end
+
+        args += %W[
+          --max-time 3
+          --header "Content-Type: text/plain; charset=utf-8"
+          --header "Accept: application/json"
+          --header "Authorization: Token #{TOKEN}"
+          --data-raw '#{category},#{tags.join(",")} count=1i #{Time.now.to_i}'
+        ]
+
+        url = "https://#{HOST}/api/v2/write?bucket=#{BUCKET}&precision=s"
+        if ENV["HOMEBREW_ANALYTICS_DEBUG"]
+          puts "#{curl} #{args.join(" ")} \"#{url}\""
+          puts Utils.popen_read(curl, *args, url)
```

Comment:
Okay, somehow this command doesn't seem to run unless I run it manually (copy it from the output). Any ideas why?

Reply:
It looks like it runs for me, but just prints an "unauthorized" message. Not sure if that's intentional.

```
{"code":"unauthorized","message":"unauthorized access"}
```

Reply:
I only get an empty message, can you give me the full output?

Reply:
Sure, here's the relevant lines (I cut out the install output and the google analytics data) from a `brew install wget`:

```
/opt/homebrew/Library/Homebrew/shims/shared/curl --max-time 3 --header "Content-Type: text/plain; charset=utf-8" --header "Accept: application/json" --header "Authorization: Token bHDUBcO8rp521Y5lzCV4nrhWaYOeV-OjGeDVxv37m9Ee75CwTjir4EQr1hCHNJaEeRowiAjw3uSh9jG1Usmqfw==" --data-raw 'install,action=wget,label=macOS\ 12, ARM,version=3.5.2-144-g14d2e08,product=Homebrew count=1i 1656262956' "https://europe-west1-1.gcp.cloud2.influxdata.com/api/v2/write?bucket=analytics&precision=s"
{"code":"unauthorized","message":"unauthorized access"}
...
/opt/homebrew/Library/Homebrew/shims/shared/curl --max-time 3 --header "Content-Type: text/plain; charset=utf-8" --header "Accept: application/json" --header "Authorization: Token bHDUBcO8rp521Y5lzCV4nrhWaYOeV-OjGeDVxv37m9Ee75CwTjir4EQr1hCHNJaEeRowiAjw3uSh9jG1Usmqfw==" --data-raw 'install_on_request,action=wget,label=macOS\ 12, ARM,version=3.5.2-144-g14d2e08,product=Homebrew count=1i 1656262957' "https://europe-west1-1.gcp.cloud2.influxdata.com/api/v2/write?bucket=analytics&precision=s"
{"code":"unauthorized","message":"unauthorized access"}
```

Reply:
Here's what happens if I run the command directly, though:

```sh
$ curl --max-time 3 --header "Content-Type: text/plain; charset=utf-8" --header "Accept: application/json" --header "Authorization: Token bHDUBcO8rp521Y5lzCV4nrhWaYOeV-OjGeDVxv37m9Ee75CwTjir4EQr1hCHNJaEeRowiAjw3uSh9jG1Usmqfw==" --data-raw 'install,action=wget,label=macOS\ 12, ARM,version=3.5.2-144-g14d2e08,product=Homebrew count=1i 1656262956' "https://europe-west1-1.gcp.cloud2.influxdata.com/api/v2/write?bucket=analytics&precision=s"
{"code":"invalid","message":"at line 1:37: expected tag key after comma; got white space instead (check rejected_points in your _monitoring bucket for further information)"}
```

Looks like an issue with the `, ARM` that's part of `Utils::Analytics::os_arch_prefix_ci`. Looks like on ARM, a custom prefix, or on CI, something is added to the output in the form of `, *` which messes up the call.

If I remove the `, ARM`, and run it manually, I just get no response. However, if I override [`Utils::Analytics::arch_label`](https://github.com/Homebrew/brew/blob/38e1af857654b940f52df9b686f31b0b0cce2d17/Library/Homebrew/extend/os/mac/utils/analytics.rb#L17-L25) to always say that I'm on Intel, I still get the unauthorized error even though there's no syntax issue.